src/eckit/config/Configuration.cc

```diff
@@ -73,6 +73,10 @@ eckit::Value Configuration::lookUp(const std::string& s, bool& found) const {
     eckit::Value result = *root_;
 
     for (size_t i = 0; i < path.size(); i++) {
+        if (!(result.isMap() || result.isOrderedMap())) {
+            found = false;
+            return result;
+        }
         const std::string& key = path[i];
```

Comment:
This I understand fixed accessing sub-configurations (?) -- not related to the PR topic, but obviously welcome!

Reply:
This is in fact the one change that I really need :).
If you want a separate PR for this, that's fine. Then we can come to solution for the rest later.

Reply:
No need! It gets 6 stars out of 5 :-)